Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -7871,8 +7871,108 @@ ErrorOr<OwnPtr<CalculationNode>> Parser::parse_a_calculation(Vector<ComponentVal
     if (parsing_failed_for_child_node)
         return nullptr;
 
-    // FIXME: 6. Return the result of simplifying a calculation tree from values.
-    return single_value.release_value();
+    // 6. Return the result of simplifying a calculation tree from values.
+    return simplify_a_calculation(single_value.release_value());
+}
+
+// https://drafts.csswg.org/css-values-4/#calc-simplification
+ErrorOr<OwnPtr<CalculationNode>> Parser::simplify_a_calculation(OwnPtr<CalculationNode> root)
```

Comment:
The input and output can both be Nonnull. By step 6 of `parse_a_calculation` we know that `single_value` isn't null, and `simplify_a_calculation()` always returns a value too.